windows/readiness_healthcheck.go

```diff
@@ -0,0 +1,98 @@
+package windows
+
+import (
+	"github.com/cloudfoundry/cf-test-helpers/v2/cf"
+	"github.com/cloudfoundry/cf-test-helpers/v2/helpers"
+
+	. "github.com/cloudfoundry/cf-acceptance-tests/cats_suite_helpers"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/app_helpers"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/assets"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/logs"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/random_name"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/skip_messages"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = WindowsDescribe("Readiness Healthcheck", func() {
+	var appName string
+	var readinessHealthCheckTimeout = "25s" // 20s route emitter sync loop + 2s hc interval + bonus
+
+	BeforeEach(func() {
+		if !Config.GetReadinessHealthChecksEnabled() {
+			Skip(skip_messages.SkipReadinessHealthChecksMessage)
+		}
+		appName = random_name.CATSRandomName("APP")
+	})
+
+	AfterEach(func() {
+		app_helpers.AppReport(appName)
+		Eventually(cf.Cf("delete", appName, "-f")).Should(Exit(0))
+	})
+
+	// TODO add a ready endpoint to Nora
+	Describe("when the readiness healthcheck is set to http", func() {
+		It("registers the route only when the readiness check passes", func() {
+			By("pushing the app")
+			// FIXME use manifest and update linux
+			Expect(cf.Cf("push",
+				appName,
+				"-f", assets.NewAssets().Nora+"/../readiness_manifest.yml",
+				"-p", assets.NewAssets().Nora,
+			).Wait(Config.CfPushTimeoutDuration())).To(Exit(0))
+
+			By("verifying the app starts")
+			Eventually(func() string {
+				return helpers.CurlAppRoot(Config, appName)
+			}, Config.CfPushTimeoutDuration()).Should(ContainSubstring("hello i am nora running on"))
+
+			By("verifying the app is marked as ready")
+			// TODO: only include this when audit events are built
+			// Eventually(func() string {
+			// 	return string(cf.Cf("events", appName).Wait().Out.Contents())
+			// }).Should(MatchRegexp("app.ready"))
+
+			// Windows apps take a Loooooong time to start apparently
+			Expect(string(logs.Recent(appName).Wait(Config.CfPushTimeoutDuration()).Out.Contents())).Should(ContainSubstring("Container passed the readiness health check"))
+
+			By("triggering the app to make the /ready endpoint fail")
+			helpers.CurlApp(Config, appName, "/ready/false")
+
+			By("verifying the app is marked as not ready")
+
+			// TODO: only include this when audit events are built
+			// Eventually(func() string {
+			// 	return string(cf.Cf("events", appName).Wait().Out.Contents())
+			// }).Should(MatchRegexp("app.notready"))
+
+			Eventually(func() string {
+				return string(logs.Recent(appName).Wait().Out.Contents())
+			}, readinessHealthCheckTimeout).Should(ContainSubstring("Container failed the readiness health check"))
+
+			By("verifying the app is removed from the routing table")
+			Eventually(func() string {
```

Comment:
Should this be `Consistently` instead of `Eventually`?

Reply:
No, this was eventually to ensure that the delay between catching the readiness failure + route-emitter updating nats + gorouter. 

Arguably we could have another consistently check for the same thing after, but that would likely need to wait 40s+.

Reply:
I guess I expected the "Container failed the readiness health check" log message, or some comparable log message, to coincide with the app being removed from the routing table. Then after receiving that, using `Consistently` would make sense to check that the routing consistently doesn't work.